api/envoy/config/core/v3/health_check.proto

```diff
@@ -103,8 +112,15 @@ message HealthCheck {
     // [#not-implemented-hide:] HTTP specific payload.
     Payload send = 3;
 
-    // [#not-implemented-hide:] HTTP specific response.
-    Payload receive = 4;
+    // Specifies a list of HTTP expected responses to match in the first 1024 bytes of the response body.
+    // If it is set, both the expected response check and status code determine the health check.
+    // When checking the response, “fuzzy” matching is performed such that each binary block must be found,
+    // and in the order specified, but not necessarily contiguous.
```

Comment:
This only applies to binary blocks, but after this change there can be textual blocks (IIUC).
This line can probably be generalized as follows:
```suggestion
    // When checking the response, each payload must be matched,
    // and in the order specified, but not necessarily contiguous.
```